controllers/volumereplication_controller.go

```diff
@@ -205,33 +208,97 @@ func (r *VolumeReplicationReconciler) SetupWithManager(mgr ctrl.Manager, cfg *co
 }
 
 // markVolumeAsPrimary defines and runs a set of tasks required to mark a volume as primary
-func (r *VolumeReplicationReconciler) markVolumeAsPrimary(volumeID string, parameters, secrets map[string]string) error {
+func (r *VolumeReplicationReconciler) markVolumeAsPrimary(volumeReplicationObjectRef types.NamespacedName, volumeID string, parameters, secrets map[string]string) error {
 	c := replication.CommonRequestParameters{
 		VolumeID:    volumeID,
 		Parameters:  parameters,
 		Secrets:     secrets,
 		Replication: r.Replication,
 	}
-	force := false
-	var markVolumeAsPrimaryTasks = []*tasks.TaskSpec{
-		{
-			Name: enableVolumeReplication,
-			Task: replication.NewEnableTask(c),
-		},
-		{
-			Name: promoteVolume,
-			Task: replication.NewPromoteVolumeTask(c, force),
-		},
+
+	instance := replicationv1alpha1.VolumeReplication{}
+	err := r.Client.Get(context.TODO(), volumeReplicationObjectRef, &instance)
+	if err != nil {
+		return err
 	}
 
-	resp := tasks.RunAll(markVolumeAsPrimaryTasks)
-	// Check error for all tasks and return error
-	for _, re := range resp {
-		if re.Error != nil {
-			r.Log.Error(re.Error, "task failed", "taskName", re.Name)
-			return re.Error
+	if instance.Spec.ImageState == replicationv1alpha1.Primary &&
+		instance.Status.State != replicationv1alpha1.Replicating {
+		resp := tasks.RunAll([]*tasks.TaskSpec{
+			{
+				Name: enableVolumeReplication,
+				Task: replication.NewEnableTask(c),
+			},
+		})
+		for _, re := range resp {
+			if re.Error != nil {
+				r.Log.Error(re.Error, "task failed", "taskName", re.Name)
+				return re.Error
+			}
 		}
 	}
+	instance.Status.State = replicationv1alpha1.Replicating
+	err = r.Client.Status().Update(context.TODO(), &instance, nil)
+	if err != nil {
+		return err
+	}
+
+	if instance.Spec.ImageState == replicationv1alpha1.Primary &&
+		instance.Status.State == replicationv1alpha1.Replicating {
```

Comment:
This will be always `replicating` as you are changing the `state` at line 240. Maybe we can remove this if statement.